### ISSUE TITLE
A11y: Decorative graphics not labelled as such

### ISSUE THREAD

**Issue by 73nici** [shopware/platform#7086]
*PHP Version:* 8.2   | *Shopware Version:* 6.6.9.0  | *Affected area / extension:* Platform(Default) 

 ----

*_Actual behaviour:_*

Decorative graphics are not labelled and are read aloud by the screen reader.screen reader users have texts read aloud twice, as the product graphic labels (CSS selector .product-image) and the product title (CSS selector .product-name) are identical.

*_Expected behaviour:_*

Remove the product graphics from the accessibility tree using the aria-hidden attribute for screen readers:
```html
<img … class="product-image is-standard" … aria-hidden="true">
```

![Image](https://github.com/user-attachments/assets/3900dcfc-a21d-4e43-bfba-62bd573d8e54)

*_How to reproduce:_*

Enable ACCESSIBILITY_TWEAKS Feature-Flag:
- go to a categoy page with product listing

**Comment by shopwareBot** [comment]
[public] Shopware automated response: This issue is linked to the internal issue: **NEXT-40841**



[created from NEXT-40841, comment 541800]

**Comment by shopwareBot** [comment]
[public] 

 > Priority changed to: **Low**

 Thank you for your report. After careful consideration, we've concluded that this issue does not require immediate attention or isn't something we can address right now. We prioritize openness and clear communication with our community, so we want to be transparent. We may not resolve all reported issues, and this one might remain unresolved. To stay effective, we periodically clean our backlog, and this issue could be closed in the future.

 For now, we'll keep this ticket open to invite further feedback from the community in case we've underestimated its impact and to allow us to revisit it later.



[created from NEXT-40841, comment 542003]

**Comment by ravenokko** [comment]
Thank you for reaching out and apologies for the long overdue response.

Your example specifically targets images in product listings, and you are right, it would be very annoying to hear the same information twice with a screen reader. After verifying with the team, however, we agreed that listing images are not (categorically) decorative.

Product names are typically not sufficient as alt text unless they actively describe what is in the listing image. For example, a product named "Compact red umbrella with large black polka dots" might arguably be sufficient alt text, but a product named "Super Cool T-Shirt" definitely would not. Unless a product name can reliably convey the same visible key information as the picture (which could include size, color, texture, shape, materials, packaging, condition, accessories, or purpose to name just a few possibilities) the associated listing image cannot be decorative - e.g. the image cannot be removed from the page without losing valuable information, and requires alt text per [WCAG S.C. 1.1.1](https://www.w3.org/WAI/WCAG22/Understanding/non-text-content.html). Because Shopware has no way of predicting (or enforcing) the names of products, we cannot assume the name alone is sufficient.

If you want to experience it yourself, disable images in chrome under Settings > Privacy and Security > Site settings > Images, then open any e-commerce site.

The Shopware logic that assigns product names as a fallback for alt-text was meant to serve as a transition for merchants upgrading from older versions (as the feature is a remnant of legacy versions), particularly merchants with a large catalog of products/images in light of the EU regulations. While responsibility for the accuracy and adequacy of product content, product names, and image descriptions ultimately remains with the merchants, Shopware will be investigating solutions to assist with writing better alt text and gradually shifting away from using product names as a fallback for alt-text.

As such, we will be closing this issue. Again, thank you for raising awareness and your help making Shopware more accessible.
